SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestMultiServerUnrealComponents/SpatialTestMultiServerUnrealComponents.cpp

```diff
@@ -0,0 +1,169 @@
+		TestActor->SpawnDynamicComponents();
+
+		const bool bWrite = true;
+		ProcessActorProperties(bWrite);
+
+		AController* PlayerController = Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		AssertTrue(PlayerController->HasAuthority(), TEXT("Server requires authority over controller"));
+		TestActor->SetOwner(PlayerController);
+
+		RegisterAutoDestroyActor(TestActor);
+
+		FinishStep();
+	});
+
+	// Check on owning client that actor exists and properties are correct
+	AddStep(
+		TEXT("SpatialTestMultiServerUnrealComponents Validate Test Actor On Owning Client"), FWorkerDefinition::Client(1),
+		[this]() -> bool {
+			return TestActor != nullptr;
+		},
+		[this]() {
+			AssertFalse(TestActor->HasAuthority(), TEXT("This client shouldn't have authority"));
+			if (AssertTrue(TestActor->AreAllDynamicComponentsValid(), TEXT("All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bOwnerOnlyExpected = true;
+				const bool bHandoverExpected = false;
+				const bool bInitialOnlyExpected = true;
+				ProcessActorProperties(bWrite, bOwnerOnlyExpected, bHandoverExpected, bInitialOnlyExpected);
+			}
+
+			FinishStep();
+		});
+
+	// Check on non-auth server that actor exists and properties are correct
+	AddStep(
+		TEXT("SpatialTestMultiServerUnrealComponents Validate Test Actor On Non-Auth Server"), FWorkerDefinition::Server(2),
+		[this]() -> bool {
+			return TestActor != nullptr;
+		},
+		[this]() {
+			AssertFalse(TestActor->HasAuthority(), TEXT("This server shouldn't have authority"));
+			if (AssertTrue(TestActor->AreAllDynamicComponentsValid(), TEXT("All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bOwnerOnlyExpected = true;
+				const bool bHandoverExpected = true;
+				const bool bInitialOnlyExpected = true;
+				ProcessActorProperties(bWrite, bOwnerOnlyExpected, bHandoverExpected, bInitialOnlyExpected);
+			}
+
+			FinishStep();
+		});
+
+	// Check on non-owning client that actor exists and properties are correct
+	AddStep(
+		TEXT("SpatialTestMultiServerUnrealComponents Validate Test Actor On Non-Owning Client"), FWorkerDefinition::Client(2),
+		[this]() -> bool {
+			return TestActor != nullptr;
+		},
+		[this]() {
+			AssertFalse(TestActor->HasAuthority(), TEXT("This client shouldn't have authority"));
+			if (AssertTrue(TestActor->AreAllDynamicComponentsValid(), TEXT("All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bOwnerOnlyExpected = false;
+				const bool bHandoverExpected = false;
+				const bool bInitialOnlyExpected = true;
+				ProcessActorProperties(bWrite, bOwnerOnlyExpected, bHandoverExpected, bInitialOnlyExpected);
+			}
+
+			FinishStep();
+		});
+}
+
+void ASpatialTestMultiServerUnrealComponents::ProcessActorProperties(bool bWrite, bool bOwnerOnlyExpected, bool bHandoverExpected,
```

Comment:
Might be worth adding a few comments to explain how this function works?

Reply:
Done